test/unit/org/apache/cassandra/db/guardrails/GuardrailMinKeyspaceReplicationFactorTest.java

```diff
@@ -0,0 +1,245 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.service.ClientWarn;
+import org.assertj.core.api.Assertions;
+
+import static java.lang.String.format;
+import static org.junit.Assert.assertNotNull;
+
+public class GuardrailMinKeyspaceReplicationFactorTest extends ThresholdTester
+{
+    private static final int MINIMUM_KEYSPACE_RF_WARN_THRESHOLD = 4;
+    private static int MINIMUM_KEYSPACE_RF_FAIL_THRESHOLD = 1;
+    private static final int DEFAULT_RF = 2;
+    private static final int DISABLED_GUARDRAIL = -1;
+    private static final String WHAT = "minimum_keyspace_rf";
+    private final TriConsumer<Guardrails, Integer, Integer> setter;
+
+    public GuardrailMinKeyspaceReplicationFactorTest()
+    {
+        super(MINIMUM_KEYSPACE_RF_WARN_THRESHOLD,
+              MINIMUM_KEYSPACE_RF_FAIL_THRESHOLD,
+              Guardrails.minimumKeyspaceRF,
+              Guardrails::setMinimumKeyspaceRFThreshold,
+              Guardrails::getMinimumKeyspaceRFWarnThreshold,
+              Guardrails::getMinimumKeyspaceRFFailThreshold);
+
+        this.setter = Guardrails::setMinimumKeyspaceRFThreshold;
+    }
+
+    @Before
+    public void setupTest() throws Throwable
+    {
+        DatabaseDescriptor.setDefaultKeyspaceRF(DEFAULT_RF);
+        MINIMUM_KEYSPACE_RF_FAIL_THRESHOLD = 2;
+
+    }
+    @After
+    public void cleanupTest() throws Throwable
+    {
+        execute("DROP KEYSPACE IF EXISTS ks");
+    }
+
+    @Override
+    protected long currentValue()
+    {
+        return Long.parseLong((Keyspace.open("ks").getReplicationStrategy()).configOptions.get("datacenter1"));
+    }
+
+    @Override
+    protected List<String> getWarnings()
+    {
+        List<String> warnings = ClientWarn.instance.getWarnings();
+
+        return warnings == null
+               ? Collections.emptyList()
+               : warnings.stream()
+                         .filter(w -> !w.contains("keyspace ks is higher than the number of nodes 1 for datacenter") &&
+                                      !w.contains("When increasing replication factor you need to run a full (-full) repair to distribute the data") &&
+                                      !w.contains("keyspace ks is higher than the number of nodes 1"))
+                         .collect(Collectors.toList());
+    }
+
+    @Test
+    public void testConfigValidation()
```

Comment:
Nit: This method has multiple missed whitespaces after commas and around `+` operators.